dts/arm/ti/am62x_phycore_m4.dtsi

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (C) 2023 PHYTEC Messtechnik GmbH
+ * Author: Daniel Schultz <d.schultz@phytec.de>
+ *
```

Comment:
Do we need this separately for this platform? 
The original PR named the corresponding file `dts/arm/ti/am62x_sk_m4.dtsi` while it should have been `am62x_m4.dtsi` , can you reuse the same file?

Reply:
Actually we do not since this file should describe the M4F subsystem which is the same in all AM62x boards. Changed the name to `am62x_m4.dtsi` and update the sk board device-tree file.